### ISSUE TITLE
Secret output properties are not marked secret in the state

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-terraform-bridge#1380]
### What happened?

When working with pulumi/pulumi-aws#2791 I noticed that simply making a property Secret in SchemaInfo is insufficient to guarantee that the values of this property never appear in the plain the Pulumi state files.

It appears that we have some machinery to do this called MarkSchemaSecrets, but it is only applied in Check that is it applies to the inputs of a resource. If something is an output property that only becomes available during Create or Update, MarkSchemaSecrets does not touch it and it is then plain texted in the state file.

### Example

In AWS provider, mark `tags_all` as Secret, provision a bucket with tags set and observe that the tags_all is not secret in the state.

### Output of `pulumi about`

```
CLI          
Version      3.74.0
Go Version   go1.20.5
Go Compiler  gc

Host     
OS       darwin
Version  13.4.1
Arch     x86_64

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/t0yv0
User           t0yv0
Organizations  t0yv0, pulumi

Pulumi locates its logs in /var/folders/gk/cchgxh512m72f_dmkcc3d09h0000gp/T/ by default
```

### Additional context

N/A

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
This sounds bad on its face value... Do you think we should prioritise this?

**Comment by t0yv0** [comment]
Yes!  It's a real quick one to workaround and very much worth doing. If I don't get to it in this iteration next one is good.

**Comment by VenelinMartinov** [comment]
I believe this issue might have been fixed since. I ran the following test:

1. Removed this line https://github.com/pulumi/pulumi-aws/blob/7c79fe28e58bfce60d3d9764c1fd6e067d73eeba/provider/resources.go#L7186-L7187
2. Built the provider and sdk.
3. Created an s3 bucket with a tag.
4. Exported the state with `pulumi stack export`
5. Observed the `tagsAll` property is marked as secret and encrypted in the state.

I also did the same but with this line removed https://github.com/pulumi/pulumi-aws/blob/7c79fe28e58bfce60d3d9764c1fd6e067d73eeba/provider/resources.go#L7183
and observed that the property is no longer secret. Hence the `Secret` property in the `SchemaInfo` must be working correctly.

I'll raise a PR to remove the unnecessary `ensureTagsAllSecret` and TODO.

Let me know if I missed something.
